working/333 - shared memory multithreading/proposal.md

```diff
@@ -1583,22 +1242,13 @@ able to implement proposed semantics on top of it.
 >    other.
 > 2. It prohibits `externref` inside shareable structs
 >
-> Dart has `Object` as a base class for both shareable and non-shareable
-> classes. If a program contains `Shareable` type - such type would need to be
-> represented as a `shared` struct which means we have to mark `Object` struct
-> as `shared` as well. But this means Dart objects can no longer directly
-> contain `externref`s inside them.
+> If Dart introduces share memory multithreading it would need to mark
+> `struct` type representing `Object` as `shared`, but this means Dart objects
+> can no longer directly contain `externref`s inside them.
 >
 > Assuming that Wasm is going to move forward with type based partitioning, we
-> can still resolve this conundrum by employing a
-> [`ShareableBox`](#shareableboxt)-like wrapper, which can be implemented on top
-> of TLS storage and `WeakMap`.
->
-> An alternative could be to tweak semantics of Wasm's `externref` a bit: tag
-> `externref` with their origin and dynamically checking origin match when
-> `externref` is passed back from Wasm to the host environment (see
-> [Dynamic sharedness checks as an escape hatch](https://github.com/WebAssembly/shared-everything-threads/issues/37)
-> issue).
+> would need to resolve this conundrum by employing some sort of thread local
+> wrapper, which can be implemented on top of TLS storage and `WeakMap`.
```

Comment:
If we can have a single object representing each `Thread`, then a `ThreadLocal` is just an `Expando<Thread>`. 
That does mean not cloning `Thread` when communicating with other isolates.

Reply:
Ack. Note TLS here refers to Wasm's TLS capabilties which are part of the shared-everything-threads proposal linked above.